### ISSUE TITLE
Closing floating windows w. alt-f4 (or equivalent) is buggy, and can cause segfaults

### ISSUE THREAD

**Issue by jplloyd** [mypaint/mypaint#1050]
#### Description of the problem

When closing a floating dockable window, such as the layers panel or the tool options panel, using standard window manager shortcuts (usually alt-f4, in my case Shift+Super+C), and subsequently reactivating any tab that was part of that floating window, I have observed one of two things:

With Gtk 3.18: The tab shows up, but is now empty. The only way to recover is to restart MyPaint.

With Gtk 3.22 (appimage): MyPaint segfaults without a chance to recover (potential for lost work).

#### Basic system details

MyPaint version: 2.0.0 beta (git revision c4b2de3ad9ca8ee6445c3373df6bc49e6ea8c83c) - but this may have been around since at least 1.2.1
Operating system: Ubuntu 16.04
Desktop environment: [awesome wm](https://awesomewm.org/)

#### Steps to reproduce

1. Start MyPaint
2. Activate (e.g.) "Tool options" panel (press T, or click the cog in toolbar)
3. Drag the tab handle to disconnect the panel to its own floating window.
4. Close the floating window
5. Reactivate "Tool options" (press T twice, or click the cog twice)

----

It seems we need to handle these floating windows better.

--

**Edit: additional testing**

On Windows (Gtk 3.24.13-1), the same procedure causes a segfault as well.

On Fedora 30 (Gtk 3.24.11-1), the floating windows cannot be closed at all, so cannot reproduce.

**Comment by jplloyd** [comment]
Addressed in 78cab07922c43713ec4c16bd39363fed495f0a12

**Comment by odysseywestra** [comment]
The fixed made are causing master to not launch in linux.

```
Traceback (most recent call last):
  File "/usr/local/bin/mypaint", line 315, in <module>
    version=version,
  File "/usr/local/lib/mypaint/gui/main.py", line 225, in main
    run()
  File "/usr/local/lib/mypaint/gui/main.py", line 198, in run
    fullscreen = options.fullscreen,
  File "/usr/local/lib/mypaint/gui/application.py", line 321, in __init__
    wkspace.build_from_layout(self.preferences["workspace.layout"])
  File "/usr/local/lib/mypaint/gui/workspace.py", line 263, in build_from_layout
    set_initial_window_position(toplevel_win, toplevel_pos)
  File "/usr/local/lib/mypaint/gui/windowing.py", line 745, in set_initial_window_position
    for mon_num in xrange(screen.get_n_monitors()):
NameError: name 'xrange' is not defined
```

Funny thing it actually build, just couldn't launch, I wonder why the CI tools didn't catch this...
